src/System.Net.Http/tests/UnitTests/StreamToStreamCopyTest.cs

```diff
@@ -0,0 +1,351 @@
+        {
+            byte[] input = CreateByteArray(8192);
+            MemoryStream source = CreateSourceMemoryStream(sourceIsExposable, input);
+            var destination = new HttpContent.LimitMemoryStream(int.MaxValue);
+
+            await StreamToStreamCopy.CopyAsync(source, destination, 0x1000, disposeSource);
+
+            Assert.NotEqual(disposeSource, source.CanRead);
+            if (!disposeSource)
+            {
+                Assert.Equal(input.Length, source.Position);
+            }
+
+            Assert.Equal(input, destination.ToArray());
+            Assert.Equal(input.Length, destination.Position);
+            Assert.Equal(input.Length, destination.Length);
+        }
+
+        [Theory]
+        [MemberData(nameof(TwoBooleans))]
+        public async Task MemoryStream_To_LimitMemoryStream_EqualCapacity(bool sourceIsExposable, bool disposeSource)
+        {
+            byte[] input = CreateByteArray(8192);
+            MemoryStream source = CreateSourceMemoryStream(sourceIsExposable, input);
+            var destination = new HttpContent.LimitMemoryStream(int.MaxValue, input.Length);
+
+            await StreamToStreamCopy.CopyAsync(source, destination, 0x1000, disposeSource);
+
+            Assert.NotEqual(disposeSource, source.CanRead);
+            if (!disposeSource)
+            {
+                Assert.Equal(input.Length, source.Position);
+            }
+
+            Assert.Equal(input, destination.ToArray());
+            Assert.Equal(input.Length, destination.Position);
+            Assert.Equal(input.Length, destination.Length);
+        }
+
+        [Theory]
+        [MemberData(nameof(TwoBooleans))]
+        public async Task MemoryStream_To_LimitMemoryStream_BiggerCapacity(bool sourceIsExposable, bool disposeSource)
+        {
+            byte[] input = CreateByteArray(8192);
+            MemoryStream source = CreateSourceMemoryStream(sourceIsExposable, input);
+            var destination = new HttpContent.LimitMemoryStream(int.MaxValue, input.Length * 2);
+
+            await StreamToStreamCopy.CopyAsync(source, destination, 0x1000, disposeSource);
+
+            Assert.NotEqual(disposeSource, source.CanRead);
+            if (!disposeSource)
+            {
+                Assert.Equal(input.Length, source.Position);
+            }
+
+            Assert.Equal(input, destination.ToArray());
+            Assert.Equal(input.Length, destination.Position);
+            Assert.Equal(input.Length, destination.Length);
+        }
+
+        [Theory]
+        [MemberData(nameof(TwoBooleans))]
+        public async Task MemoryStream_To_LimitMemoryStream_SmallerCapacity(bool sourceIsExposable, bool disposeSource)
+        {
+            byte[] input = CreateByteArray(8192);
+            MemoryStream source = CreateSourceMemoryStream(sourceIsExposable, input);
+            var destination = new HttpContent.LimitMemoryStream(int.MaxValue, 1024);
```

Comment:
I mean copying past MaxSize.  Since the new stream copy pokes about with its innards it seems like a valid test to me to ensure that if it threw on too-much-copying before that it still throws now.  (If it didn't throw, then I guess I just misunderstand its purpose)